sse.go

```
@@ -0,0 +1,144 @@
+		ctx.ResponseWriter().Header().Set("Connection", "keep-alive")
+		ctx.ResponseWriter().Header().Set("X-Accel-Buffering", "no")
+
+		sse := &Connection{
+			Options: newOptions(options),
+			context: ctx,
+			Sender:  makeChanOfType(reflect.TypeOf(bind), 1),
+		}
+		ctx.Set(reflect.ChanOf(reflect.SendDir, sse.Sender.Type().Elem()), sse.Sender)
+
+		go sse.handle()
+
+		ctx.Next()
+	}
+}
+
+// newOptions creates new default options and assigns any given options.
+func newOptions(options []*Options) *Options {
+	if len(options) == 0 {
+		return &Options{
+			PingInterval: defaultPingInterval,
+		}
+	}
+	return options[0]
+}
+
+// startTicker starts the ticker used for pinging the client.
+func (c *Connection) startTicker() {
+	c.ticker = time.NewTicker(c.PingInterval)
+}
+
+// stopTicker stop the ticker used for pinging the client.
+func (c *Connection) stopTicker() {
+	c.ticker.Stop()
+}
+
+func (c *Connection) write(msg string) {
+	_, _ = c.context.ResponseWriter().Write([]byte(msg))
+}
+
+func (c *Connection) flush() {
+	c.context.ResponseWriter().Flush()
+}
+
+const (
+	senderSend = iota
+	tickerTick
+	timeout
+)
+
+func (c *Connection) handle() {
+	c.startTicker()
+	defer func() { c.stopTicker() }()
+
+	c.write(": ping\n\n")
+	c.write("events: stream opened\n\n")
+	c.flush()
+
+	cases := make([]reflect.SelectCase, 3)
+	cases[senderSend] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: c.Sender, Send: reflect.ValueOf(nil)}
+	cases[tickerTick] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(c.ticker.C), Send: reflect.ValueOf(nil)}
+	cases[timeout] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(time.After(time.Hour)), Send: reflect.ValueOf(nil)}
+
+L:
+	for {
+		chosen, message, ok := reflect.Select(cases)
+		switch chosen {
+
+		case senderSend:
+			if !ok {
+				// Sender channel has been closed.
+				return
+			}
+
+			c.write("data: ")
+			evt, _ := json.Marshal(message.Interface())
```

What's the worst the thing that could happen when we ignore the error here?